### ISSUE TITLE
Unable to reach AUC results showed in the paper on the UCF-Crime testing set

### ISSUE THREAD

**Issue by luca-zanella-dvl** [fjchange/MIST_VAD#6]
Hi,

First, congratulations for the great paper and thank you for publishing the code.

Unfortunately, I'm not able to reproduce the results described in the paper on the UCF-Crime dataset. As described in your testing guidelines, I did the following steps:
-  I downloaded the UCF-Crime dataset and I transformed the testing set into a single h5py file (for doing that I had to slightly modify your `utils/make_h5.py` script, see #5). I used the default value 16 for the `segment_len` parameter.
- I dowloaded the pretrained weights you uploaded on OneDrive.
- I modified the paths in the `configs/constant.py` file to adapt to my project.
- I slightly modified your `testing/test.py`. Particularly, I wrapped the `model` object using `torch.nn.DataParallel`, I used the key `model` instead of `state_dict` when loading the model’s parameter dictionary, and I deleted the first occurrence of `return eval(total_scores, total_labels)` which was making the second occurrence unreachable. I might do a pull request if you are interested.
- I tested the model by running `python test.py --gpus 0 --MODEL UCF_I3D` and `python test.py --gpus 0 --MODEL UCF_C3D` (I slightly modified the `configs/options.py` file to accept the `--gpus` argument).

With these modifications, I managed to run the inference using python=3.8, pytorch=1.7.0+cu110, and torchvision==0.8.1+cu110 but I cannot get AUC results close to yours.

Do you have any suggestion on how to reproduce the results on the UCF-Crime testing set as described in your paper?

Many thanks in advance.

**Comment by fjchange** [comment]
Thanks for report some bugs. I have updated the codes and re-uploaded the weights which have been checked  last night. Hope it help.

**Comment by luca-zanella-dvl** [comment]
Many thanks for your effort. I truly appreciate it! Now, I can reproduce the results reported in table 1 of the paper on the UCF-Crime testing set. Here is a list of minor changes I had to make: 
- there is a single quote missing on the third line `sys.path.append('..')`
- in the `eval_UCF` function, instead of passing the parameter `anno.tolist()` to `n_scores.extend`, I pass `score`
- the return statement of `eval_UCF` is missing a third parameter, namely `n_scores`
- in the `eval` function, I convert `[0]*len(n_scores)` to a NumPy array before feeding it to the `cal_false_alarm` function, so that a single integer value can then be subtracted from it.

Also, for those people who are trying to reproduce the results on the UCF-Crime test set, remember to replace `path='Normal/'+path.strip().split('/')[-1]` with `path='Testing_Normal_Videos_Anomaly/'+path.strip().split('/')[-1]` in `make_h5.py`.

I tested your code using python=3.8, torch==1.9.0+cu111, and torchvision==0.10.0+cu111. As I can reproduce the results, I am closing the issue. 

Thanks again!
